src/test/suite/extension.test.ts

```diff
@@ -0,0 +1,138 @@
+/* eslint-disable @typescript-eslint/no-empty-function, @typescript-eslint/no-explicit-any */
+import * as assert from 'assert'
+
+// You can import and use all API from the 'vscode' module
+// as well as import your extension to test it
+import * as vscode from 'vscode'
+import * as vsflux from '../../extension'
+
+class FauxEnvironmentVariableCollection {
+    persistent = false
+    replace(_variable : string, _value : string) : void { }
+    append(_variable : string, _value : string) : void { }
+    prepend(_variable : string, _value : string) : void { }
+    get(_variable : string) : vscode.EnvironmentVariableMutator | undefined { return undefined }
+    forEach(_callback : (variable : string, mutator : vscode.EnvironmentVariableMutator, collection : vscode.EnvironmentVariableCollection) => any, _thisArg ?: any) : void { }
+    delete(_variable : string) : void { }
+    clear() : void { }
+}
+
+class FauxMemento {
+    private data : { [key : string] : any }
+
+    constructor() {
+        this.data = {}
+    }
+
+    keys() : string[] {
+        return Object.keys(this.data)
+    }
+
+    get<T>(key : string) : T | undefined;
+    get<T>(key : string, defaultValue : T) : T;
+    get<T>(key : string, defaultValue ?: T) : T | undefined {
+        if (this.keys().includes(key) || defaultValue === undefined) {
+            return this.get(key) as T
+        } else {
+            return defaultValue
+        }
+    }
+
+    update(key : string, value : any) : Thenable<void> {
+        return new Promise((resolve, _reject) => {
+            this.data[key] = value
+            resolve()
+        })
+    }
```

Comment:
Why not? Ditto for the other `Thenable`/`Promise` returning methods
```suggestion
    async update(key : string, value : any) : Promise<void> {
        this.data[key] = value;
    }
```

Reply:
Mostly because I'm just dummy-implementing an interface. I made that one functional before realizing how many I'd have to implement. If I were to re-work it today, I'd just take the functionality out, because I don't care about it.

The goal of this work, while yes, it has tests, was more about having some way to say "is the extension so broken it won't load?" We released a broken extension once that was like that, and I'd prefer to have something keep us from doing that in the future. As it stands, I still can't run these tests in CI, so I'm not going to get deep/extensive in the testing until I figure that part out.